hack/verify-checkoutput.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script does a fast type check of kubernetes code for all platforms.
+# Usage: `hack/verify-checkoutput.sh`.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+source "${KUBE_ROOT}/hack/lib/init.sh"
+
+kube::golang::verify_go_version
+
+cd "${KUBE_ROOT}"
+
+make --no-print-directory -C "${KUBE_ROOT}" generated_files
+
+echo ${KUBE_ROOT}
+ret=0
+hack/run-in-gopath.sh \
+    go run test/checkoutput/main.go "$@" || ret=$?
+if [[ $ret -ne 0 ]]; then
+  echo "!!! checkoutput has failed. This mostly implies a parsing issue" >&2
```

Comment:
Why does that imply a parsing issue?

We want a non-zero return code for both invalid input and test failures.

Reply:
This comment was made when it originally took a single Markdown file as input. I forgot to update it.

Reply:
Fixed.